test/builder/BridgingLogic.t.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.27;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import {QuarkBuilderTest} from "test/builder/lib/QuarkBuilderTest.sol";
+
+import {AcrossActions} from "src/AcrossScripts.sol";
+import {CCTPBridgeActions} from "src/BridgeScripts.sol";
+import {Multicall} from "src/Multicall.sol";
+import {TransferActions} from "src/DeFiScripts.sol";
+import {WrapperActions} from "src/WrapperScripts.sol";
+import {TransferActionsBuilder} from "src/builder/actions/TransferActionsBuilder.sol";
+import {Actions} from "src/builder/actions/Actions.sol";
+import {Accounts} from "src/builder/Accounts.sol";
+import {Across} from "src/builder/BridgeRoutes.sol";
+import {CodeJarHelper} from "src/builder/CodeJarHelper.sol";
+import {Paycall} from "src/Paycall.sol";
+import {PaycallWrapper} from "src/builder/PaycallWrapper.sol";
+import {PaymentInfo} from "src/builder/PaymentInfo.sol";
+import {QuarkBuilder} from "src/builder/QuarkBuilder.sol";
+import {QuarkBuilderBase} from "src/builder/QuarkBuilderBase.sol";
+import {Quotecall} from "src/Quotecall.sol";
+import {YulHelper} from "test/lib/YulHelper.sol";
+
+import {AcrossFFI} from "test/builder/mocks/AcrossFFI.sol";
+
+contract BridgingLogicTest is Test, QuarkBuilderTest {
+    function setUp() public {
+        // Deploy mock FFI for calling Across API
+        AcrossFFI mockFFI = new AcrossFFI();
+        vm.etch(Actions.ACROSS_FFI_ADDRESS, address(mockFFI).code);
```

Comment:
Well, I was about to write a long rant about how to test this and you nailed it here. Awesome.